Babylon/commands/api/solutions/create.py

```diff
@@ -37,35 +39,39 @@
 )
 @option("--select", "select", is_flag=True, default=True, help="Save this new solution in configuration")
 @argument("name", type=QueryType())
-@inject_context_with_resource({"api": ['url', 'organization_id'], 'acr': ['simulator_repository', 'simulator_version']})
-def create(ctx: Context, context: Any, azure_token: str, name: str, solution_file: pathlib.Path,
+@retrieve_state
+def create(state: Any, azure_token: str, organization_id: str, solution_name: str, solution_file: pathlib.Path,
            select: bool) -> CommandResponse:
     """
-    Register new solution
+    Register a new solution
     """
-    check_ascii(name)
+    state = state['state']
+    state['api']['organization_id'] = organization_id or state['api']['organization_id']
+
+    check_ascii(solution_name)
     path_file = f"{env.context_id}.{env.environ_id}.solution.yaml"
     t_file = solution_file or env.working_dir.payload_path / path_file
     if not t_file.exists():
         logger.error(f"No such file: '{basename(t_file)}' in .payload directory")
         sys.exit(1)
```

Comment:
IIUC the only way to provide the payload file will be the path given in the command call and` f"{env.context_id}.{env.environ_id}.solution.yaml"` file will be deleted, so I think we can delete these checks